src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -211,3 +243,60 @@ def _extract_rail(string: str, positive_rail: str, negative_rail: str) -> Option
     elif negative_pos != -1:
         return negative_rail
     return None
+
+
+def retrieval_concat_and_truncate(values, encoding_name="gpt-4", max_tokens=7100):
```

Comment:
let's extract into a separate PR - it also feels like a pretty brute force solution that can affect the output - what if the context is at the end of the text? I think we should consider the reduce approach instead.